### PR TITLE
Cluedo mystery

### PR DESCRIPTION
**Pull request by thereselinden**
This week I managed to completed blue, red and black level. I'm very happy that I managed this. 
Focused very little on the styling. If I had more time I would have tried to make my "resume" button appear after I had clicked "reveal" but did not have time for that this week.

### CODE REVIEW

File: code/clue.js [hunk 1]
```diff
@@ -1,59 +1,401 @@
-// STEP 1 - CREATE OBJECTS FOR ALL THE SUSPECTS, SOMETHING LIKE THIS:
+//variables to prevent clicking cards in wrong order. 
+let killerIsPicked = false;
+let weaponIsPicked = false;
+let roomIsPicked = false;
```
Comment (Andrea-Osmar) [inline code/clue.js]: Great feature!

File: code/clue.js [hunk 2]
```diff
@@ -1,59 +1,401 @@
...
+  name: 'guest House',
+  id: 'guest House'
+}
+
+const patio = {
+  name: 'patio',
+  id: 'patio'
+}
+ 
+
```
Comment (Andrea-Osmar) [inline code/clue.js]: no consistency in line breaking here

File: code/clue.js [hunk 3]
```diff
@@ -1,59 +1,401 @@
...
-// This function will be invoked when you click on the killer card.
+
+//object that keeps mystery 
+const mystery = {
+  killer: null,
+  weapon: null,
+  room: null
+};
+
+
```
Comment (Andrea-Osmar) [inline code/clue.js]: why so much space here?

File: code/clue.js [hunk 4]
```diff
@@ -1,59 +1,401 @@
...
+      mystery.killer = randomSelector(suspects)
+      document.getElementById('killerName').style.background = mystery.killer.color
+      document.getElementById(
+        'killerName'
+      ).innerHTML = `Name: ${mystery.killer.firstName} ${mystery.killer.lastName}`
+      document.getElementById('killerAge').innerHTML = (`Age: ${mystery.killer.age}`)
+      document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`
+      document.getElementById('killerDescription').innerHTML = `Descripton: ${mystery.killer.description}`
+      document.getElementById('killerFavouriteWeapon').innerHTML = `Preferred weapon: ${mystery.killer.favouriteWeapon}` //red level to check that the fav-weap changes by using function "shuffleFavouriteWeapon"
+      document.getElementById('killerImage').src = mystery.killer.image;
```
Comment (Andrea-Osmar) [inline code/clue.js:289]: Nice with the comment that the code is red level!

File: code/clue.js [hunk 5]
```diff
@@ -1,59 +1,401 @@
...
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  if (!killerIsPicked) {
+    document.getElementById('loaderKiller').style.opacity = 1
+    setTimeout(function () {
+      document.getElementById('loaderKiller').style.opacity = 0
+      shuffleFavoriteWeapon();
+      mystery.killer = randomSelector(suspects)
+      document.getElementById('killerName').style.background = mystery.killer.color
+      document.getElementById(
```
Comment (Andrea-Osmar) [inline code/clue.js:282]: some indentation would be good here  :)

File: code/clue.js [hunk 6]
```diff
@@ -1,59 +1,401 @@
...
+  suspects.forEach(suspect => {
+    let randomWeapon = randomSelector(weapons).id;
+    suspect.favouriteWeapon = randomWeapon
+  });
+};
+
+
+//function to increase the probabilty of favourite weapon to be picked. 
+const chanceOfFavouriteWeapon = () => {
+  const favWeapon = weapons.find(weapon => weapon.id === mystery.killer.favouriteWeapon);
```
Comment (Andrea-Osmar) [inline code/clue.js:335]: really clean code! Looks great!

File: code/clue.js [hunk 7]
```diff
@@ -1,59 +1,401 @@
...
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+//Function will be invoked when you click 'resume game' button
+const restartGame = () => {
+  let confirmation = confirm(`Click OK if you want to restart the game?`);
+
```
Comment (Andrea-Osmar) [inline code/clue.js]: Great feature with the resume game button! 🥇

File: code/index.html [hunk 8]
```diff
@@ -4,34 +4,63 @@
     <meta charset="UTF-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <link href="https://fonts.googleapis.com/css2?family=Amatic+SC:wght@700&display=swap" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css2?family=Amatic+SC:wght@700&family=Ubuntu+Mono:wght@700&display=swap" rel="stylesheet">
+    <link
+      rel="icon"
+      type="image/png"
+      href="./favicon-icon.png"
```
Comment (Andrea-Osmar) [inline code/index.html:12]: Stylish favicon ;)

File: code/style.css [hunk 9]
```diff
@@ -1,58 +1,137 @@
...
+  padding: 0;
+  font-family: 'Amatic SC', cursive;
+  font-family: 'Ubuntu Mono', monospace;  
+  background-image: radial-gradient(#174959, #0B2F44);
+
+}
+
+h1 {
+  color: white;
+  text-align: center;
```
Comment (Andrea-Osmar) [inline code/style.css:11]: Nice header!

File: code/style.css [hunk 10]
```diff
@@ -1,58 +1,137 @@
 body {
   margin: 0;
-  font-family: "Courier New", Courier, monospace;
+  padding: 0;
+  font-family: 'Amatic SC', cursive;
+  font-family: 'Ubuntu Mono', monospace;
```
Comment (Andrea-Osmar) [inline code/style.css:5]: Awesome that you are using the radial-gradient style!

File: code/style.css [hunk 11]
```diff
@@ -1,58 +1,137 @@
...
 }
 
-.killer-deck {
-  width: 250px;
-  height: 400px;
-  background: #000;
-  text-align: center;
-  padding-top: 30px;
-  border-radius: 16px;
+
```
Comment (Andrea-Osmar) [inline code/style.css]: why so much space here?

File: code/style.css [hunk 12]
```diff
@@ -1,58 +1,137 @@
...
+  opacity: 0;
+  position: absolute;
+  top: 40%;
+  left: 37%;
+  border: 10px solid #f3f3f3;
+  border-radius: 50%;
+  border-top: 10px solid #234668;
+  width: 50px;
+  height: 50px;
+  -webkit-animation: spin 4s linear infinite;
```
Comment (Andrea-Osmar) [inline code/style.css:116]: the animation here is great! good job!

**Review by Andrea-Osmar — COMMENTED** [review]
Really great job!
You keep the code clean and dry and make it simple to follow.
You had great comments in the Javascript explaining the code, wouldn't mind seeing comments in the css as well.
You could work in consistency in line-breaks and indentation. 

Overall I think you did amazing!! It was really cool that you managed to do the red level and black level!! 🥇

File: code/clue.js (hunk 3 above)
Comment (thereselinden) [reply code/clue.js]: Thank you for the input, I will take a look at this throughout the code =)

File: code/clue.js (hunk 5 above)
Comment (thereselinden) [reply code/clue.js:282]: Great, thank you for the input!

File: code/index.html (hunk 8 above)
Comment (thereselinden) [reply code/index.html:12]: Thank you =)

File: code/style.css (hunk 11 above)
Comment (thereselinden) [reply code/style.css]: Thank you! I will take a look at that!

File: code/clue.js (hunk 1 above)
Comment (EmOlsz) [reply code/clue.js]: Yes!

File: code/clue.js [hunk 13]
```diff
@@ -1,59 +1,374 @@
...
+const spa = {
+  name: 'spa',
+  id: 'spa'
 }
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const livingRoom = {
+  name: 'living Room',
+  id: 'living Room'
```
Comment (EmOlsz) [inline code/clue.js]: id's should have single words connected by `-`, so :
```
id: 'living-room'
```

File: code/clue.js [hunk 14]
```diff
@@ -1,59 +1,374 @@
...
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+//object that keeps mystery 
+const mystery = {
+  killer: null,
+  weapon: null,
+  room: null
```
Comment (EmOlsz) [inline code/clue.js:270]: Very good that it's initialised with `null` values, that's the most proper approach:)

File: code/clue.js [hunk 15]
```diff
@@ -1,59 +1,374 @@
...
+};
 
-// This function will be invoked when you click on the killer card.
+// This function will be invoked when you click the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  if (!killerIsPicked) {
```
Comment (EmOlsz) [inline code/clue.js]: Great Therese, that's the best and cleanest way of writing this statement

File: code/clue.js [hunk 16]
```diff
@@ -1,59 +1,374 @@
...
+    document.getElementById('loaderKiller').style.opacity = 1
+    setTimeout(function () {
+      document.getElementById('loaderKiller').style.opacity = 0
+      shuffleFavoriteWeapon();
+      mystery.killer = randomSelector(suspects)
```
Comment (EmOlsz) [inline code/clue.js]: you are forgeting about semicolons from time to time

File: code/clue.js [hunk 17]
```diff
@@ -1,59 +1,374 @@
...
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+// This function will be invoked when you click the weapon card.
+const pickWeapon = () => {
+  if (killerIsPicked && !weaponIsPicked) {
+    document.getElementById('loaderWeapon').style.opacity = 1
+    setTimeout(function() {
```
Comment (EmOlsz) [inline code/clue.js]: `setTimeout(() => {...}, 2500)` would be better approach, we shouldn't use `function` keyword a lot

File: code/clue.js [hunk 18]
```diff
@@ -1,59 +1,374 @@
...
+//This function will be invoked when you click  the room card.
+const pickRoom = () => {
+  if (weaponIsPicked && !roomIsPicked) {
+    document.getElementById('loaderRoom').style.opacity = 1
+    setTimeout(function() { 
+      document.getElementById('loaderRoom').style.opacity = 0
+      let chanceForSuspectedRoom = chanceOfRoom();
+      mystery.room = randomSelector(chanceForSuspectedRoom) 
+      document.getElementById('roomName').innerHTML = `Where murder was commited: ${mystery.room.name}`
+      roomsIsPicked = true
```
Comment (EmOlsz) [inline code/clue.js]: Semantically, it should rather be `isRoomPicked` than `roomIsPicked`

File: code/clue.js [hunk 19]
```diff
@@ -1,59 +1,374 @@
...
+    suspect.favouriteWeapon = randomWeapon
+  });
+};
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+//function to increase the probabilty of favourite weapon to be picked. 
+const chanceOfFavouriteWeapon = () => {
+  const favWeapon = weapons.find(weapon => weapon.id === mystery.killer.favouriteWeapon);
+  const favWeaponArray = new Array(7).fill(favWeapon);
+  const calculatedWeapon = [...weapons, ...favWeaponArray];
```
Comment (EmOlsz) [inline code/clue.js:337]: Great :)

File: code/clue.js [hunk 20]
```diff
@@ -1,59 +1,374 @@
...
+  const calculatedRooms = [...rooms, ...suspectedRoomArray];
+  return calculatedRooms;
+};
+
+//revealMystery  will be invoked when you click the 'reveal the crime' button.
+const revealMystery = () => {
+  if (
+    mystery.killer === null ||
+    mystery.weapon === null ||
+    mystery.room === null
```
Comment (EmOlsz) [inline code/clue.js]: Or we could write it shorter way :
```
if (!mystery.killer || !mystery.weapon || !mystery.room) { ... }
```

File: code/clue.js [hunk 21]
```diff
@@ -1,59 +1,374 @@
...
+    }
+}
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+//Function will be invoked when you click 'resume game' button
+const restartGame = () => {
+  let confirmation = confirm(`Click OK if you want to restart the game?`);
+  
+  if (confirmation === true) {
```
Comment (EmOlsz) [inline code/clue.js]: `if (confirmation) { ... }`

**Review by EmOlsz — CHANGES_REQUESTED** [review]
Great, clean project Therese! I'm happy to see such advanced, yet consequent JS code :) You did really good job this week.
There are some things you need to fix, take a look on my comments and provide changes. I hope they will help you improve even more :)

File: code/clue.js (hunk 13 above)
Comment (thereselinden) [reply code/clue.js]: To be consistent throughout the code, I updated room IDs with kebabCase. Don't know the "best" approach for IDs, have gotten different answer depending on who I ask. Does it matter? (of course if I'm consistent).

File: code/clue.js (hunk 15 above)
Comment (thereselinden) [reply code/clue.js]: Thank you!

File: code/clue.js (hunk 16 above)
Comment (thereselinden) [reply code/clue.js]: Thanks for the input, I have updated semicolons.

File: code/clue.js (hunk 17 above)
Comment (thereselinden) [reply code/clue.js]: Correct, thank you. I have changed that no to arrow function instead.

File: code/clue.js (hunk 18 above)
Comment (thereselinden) [reply code/clue.js]: Correct, it makes a lot more sense. I have changed that now.

File: code/clue.js (hunk 20 above)
Comment (thereselinden) [reply code/clue.js]: Thank you, a lot cleaner! I have changed that now.

File: code/clue.js (hunk 21 above)
Comment (thereselinden) [reply code/clue.js]: Great, I have changed this to!

**Comment by thereselinden** [comment]
Thank you for your feedback @EmOlsz , really appreciated it. 
I have pushed the new version to git now. 
Have a nice weekend :)

**Review by EmOlsz — APPROVED** [review]

**Pull request closed without merging**